### DISCUSSION TITLE
Proud to say SSW RulesGPT is released! Give us your feedback...

### DISCUSSION THREAD

**Discussion by adamcogan** [SSWConsulting/SSW.Rules.GPT.Blazor#46] [Announcements]
Try it out https://rulesgpt.ssw.com.au/

Did you find it useful?
What question did you ask it?
-a

**Comment by jernejk** [comment]
I used the bot several times when it was hosted on Calum's Azure, and it found me what I was looking for when Google failed me. 😊

I used it on "How do I do pagination in EF Core?" and seemed to find the right rule. 😊

There is a bug in the rendering code, though.

![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/5943653/d604001a-3703-45fd-be07-f74a3e54f09c)

Nice that it mentions to not include explicit joins. 😊
It would be great if it would include the main rule URL as well, I noticed most of the prompts don't do that. 🤔

**Comment by jimmidier** [comment]
- 😊RuleGPT could find the correct rule with source link if asking properly
- 😊RuleGPT was able to beat Google on certain questions
- 😓For some reason RuleGPT could understand keyword phrases just like Google, but easily ended up in giving deceptive answers.
![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/16027480/086ab326-fd4a-4270-bfc6-2188384b9bb7)

**Comment by tiagov8** [comment]
Hey team, 

I asked for an email template and the answer was satisfactory, however it was NOT following one of our internal standards: https://ssw.com.au/rules/placeholder-for-replaceable-text/

1. Is this something we can/should fix?

<img width="395" alt="Screenshot 2023-05-18 at 2 45 52 PM" src="https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/12601228/98b79769-5555-4209-93d5-05ab8bed0d58">

**Comment by tiagov8** [comment]
❌ UI - Button over answer

Hi team,

1. Can we move "Cancel request" button so it's never over the answer text?  
At the end the button goes away, but I believe most people will be reading the answer as it's appearing on the screen... 

<img width="1264" alt="Screenshot 2023-05-18 at 2 49 07 PM" src="https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/12601228/20586792-dae1-48fc-86d1-3391ce7f3801">

**Comment by tiagov8** [comment]
Got an wrong answer and a broken link when I asked about RulesGPT tagline.

- Wrong answer given: 

   > RulesGPT's tagline is "The AI writing assistant for software developers".

- Broken link given: 

   https://www.ssw.com.au/rules/rulesgpt

---

BTW - I am NOT a big fan of "Keeping Order with Intelligence". I feel the term sounds a bit authoritarian 

---

<img width="1511" alt="Screenshot 2023-05-18 at 2 56 17 PM" src="https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/12601228/39d33620-be26-4dc0-b028-7a8c09ad6f9e">

**Comment by matt-goldman** [comment]
The response was pretty good. It got the context right which Google failed to do, and made a suggestion to avoid the problem in future as well as how to fix it. But it replied with the wrong template. Also would be cool if we could preserve the template formatting.
<img width="899" alt="image" src="https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/19944129/d380b878-0833-4344-800f-d5cad9059ced">

**Comment by leehawkinsssw** [comment]
I like the idea (and think I proposed it in a CTF response some time ago) and it's a useful application of GPT. It appears to work better than the simple Google search for more complicated questions I might want to ask of the Rules. 

If you need a tester as you build this out, count me in!🪲

**Comment by ozdemir-mehmet** [comment]
Google Bard gives a way better answer than RulesGTP (at the moment), more development needed.

**Comment by jakebayliss** [comment]
Awesome idea - I think this was one of the ideas that came out of a brainstorming day so cool to see our ideas come to life. I will likely use this instead of searching the rules now :)

**Comment by brydeno** [comment]
It didn't give me any rules reference, nor did it use any of the rules about SQL indexing when asked about creating a SQL index.

**Reply by Marxoz** [reply]
Strange, I asked "How should I use indexes with sql" and it did

**Comment by tkapa** [comment]
Asking the robot what the best way to use the robot is didn't link me to the cheat sheet

**Comment by vladislav-kir** [comment]
RulesGPT return incorrect links to the rules... For that reason, I would rather just use Google

**Reply by bahjatalaadelssw** [reply]
Prompt: Give me a list of all the rules that pertain to video production

Result: It missed a lot - Video editing terms?? and I asked for "all" and it decided to give me "some" most disagreeable.

<img width="911" alt="Screenshot 2023-05-19 at 8 46 52 am" src="https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/108774015/5506ca57-244c-4428-ac5d-bf357ddc7ee1">

**Reply by jontygardner** [reply]
The way the prompt phrases matters looks like, got a slightly better response:

![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/97415708/3e213a37-5b73-42da-bf83-c02f1f0c1357)

**Comment by danielmackay** [comment]
Correct answer to the question.

![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/2636640/4ae45f6b-910c-49c4-a87a-ca4fef691d60)

However, I did try this last night and the answer was incorrect (even though it pointed to the correct rule), and the link was broken.

**Comment by ncn-ssw** [comment]
An excellent, promising start that requires refinement to improve the consistency and accuracy of its answers. To trial its flexibility, I asked it "Can you provide a set of questions and answers regarding the use of emojis? Be as concise as possible." The result was good.

> Absolutely! Here are some common questions and answers regarding the use of emojis:
>
> Q: Why should I use emojis in my interface?
> A: Adding emojis can make your interface more visually interesting and easier to scan quickly, helping users to better understand and remember important information. 🤔
>
> Q: How can I use emojis effectively in my interface?
> A: Use emojis sparingly and purposefully, making sure they are relevant to the content and don't create visual clutter. Also, be sure to leave a space between the emoji and the text for better readability. 🙌
>
> Q: Can I use emojis in professional or technical contexts?
> A: Yes, using emojis can be a fun and easy way to make technical or professional content more accessible and engaging, as long as they are used appropriately and not overused. 💼
> 
> Q: When should I avoid using emojis?
> A: Avoid using emojis excessively or in situations where they may be inappropriate or unprofessional, such as in formal emails or legal documents. 🚫
> 
> Q: How can I make sure my use of emojis is consistent and effective?
> A: Following the DRY principle (Don't Repeat Yourself) can help you avoid excessive repetition of emojis, making your content more visually interesting and easy to read. Also, consider using icons in addition to emojis to help convey meaning. 🤓
>
> I hope this helps! Let me know if you have any other questions. 😊

**Comment by jeoffreyfischer** [comment]
Good answer!
![rulesGPT](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/106663901/e6e42d87-10e6-45e9-b1d1-162bd771dce6)

**Reply by JeanThirion** [reply]
I'll never ask you again !

**Comment by cvionnet** [comment]
The response and referral links are good!
I'd like to be able to edit a question

![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/46954368/e6de759f-6e8a-418e-a556-6b7be00171c1)

**Comment by calumjs** [comment]
I think it's really cool - and I would encourage anyone with API access to gpt-4 to put in their own key and change the model through the cog wheel in the top right. The answers are much higher quality with less making up fake URLs etc.

**Comment by lukecookssw** [comment]
It's pretty good, but the fact that it makes up non-existent rules is a bummer...

**Comment by jaydenalchin** [comment]
Awesome idea but RulesGPT didn't give me a better answer than google. Linked rules less relevant than the one I had in mind when I asked one of my test questions (gave me dev info vs design info even though I did specify). The name & tagline could be improved but the UI is nice.

**Comment by wicksipedia** [comment]
Asked “how do I quote people in an email”

would’ve expected an example instead of the description of what it should be.

also, I would’ve expected the answer then follow up with the sources it cited, not the other way around

UI looks like it supports light/dark mode but started in light mode (my phone was in dark mode)

**Comment by MattParkerDev** [comment]
A very useful tool even in its early stages!

**Comment by Marxoz** [comment]
It is a good start!! even if it is just getting the URL for the applicable rule. I tried a few simple questions like "How should I use indexes with sql" and got the correct results.

**Comment by AlexRothwell** [comment]
This is an application of these large language models that I've been very keen to see. It'd be great if it could be tweaked to stick to only provide existing sources instead of making things up

**Reply by leehawkinsssw** [reply]
> It'd be great if it could be tweaked to stick to only provide existing sources instead of making things up

It's a bit strange that we really need to make this explicit, but it is what it is.

**Comment by tomek-i** [comment]
![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/3699937/14ad3a48-5db2-4de9-9168-af9628cc3d28)

**Comment by SamBWagner** [comment]
I like it. It's a simple project that leverages the llm well. I also like that the application is simple and just does what it says on the tin. 

That being said... I did ask it this and got the answer to store docs in Rules... 

![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/66365977/f95fc59e-7faa-4759-b8ea-2b1a95b040a1)

**Comment by kikibianc** [comment]
I really like that RulesGPT gives links and email templates, even when asking about non-rules related stuff :)

**Comment by AshAnil** [comment]
RulesGPT follows our SSW rules precisely and reliably. It not only gives answers but also provides reference links for more information ⭐

**Comment by apforsyth** [comment]
Great tips Luke!  This is a really clear and powerful explanation for specific time saving techniques for developers

**Comment by tombui99** [comment]
This is a very neat tool, definitely prefer this over Googling for stuff on SSW Rules

![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/67776356/1adf2fa7-7e65-49d2-882f-caa124bc311b)

**Comment by samsmithssw** [comment]
It got a bit confused and also didn't implement the rule that it referenced
![Screenshot 2023-05-19 at 09 50 51](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/115435510/8ceed2a4-9c2b-4510-931d-71c582bc1135)

**Comment by zacharykeeping** [comment]
Very cool tool and a good implementation of this sort of tech, and seems to work pretty well so far. I'm interested to see how this develops!

**Comment by Levijj22** [comment]
Awesome tool! 
Well done @calumjs, @MattParkerDev, @jackreimers and @Harry-Ross!

This is fantastic for sales! We are already getting enquires for implementing GPT into companies, so I look forward to using RulesGPT as an example of SSW's awesome skills.

**Comment by christoment** [comment]
Background context: I know this exists somewhere, but I couldn't figure out in which rule!

Question is: `When I send v2 email, do I keep the old email?` 

Answer:
![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/35278147/1fe54a29-ddc9-40a7-8249-a32e38b5c035)


Love it! - Although the cupcake emoji is not needed...

**Reply by calumjs** [reply]
Cupcake? I think you'll find that's a robot 😅

**Comment by AntPolkanov** [comment]
Not bad.
Good:
+ Gives the correct link
+ Gives meaningful suggestions

Bad:
- Proposes to use MSTest... 
- The 5th suggestion is obscure 

![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/123032112/9cbce792-a2c2-420d-bada-b8ddace5aa6d)

**Comment by stephanfakossw** [comment]
I've been cheeky and asked the AI overlord to teach me how to enter overtime... It returned the below
![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/96279310/bf1b35a1-9a76-41de-9bc1-3b55f4357d3b)

As this is obviously made up, I've followed the SSW Rules link to find the below
![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/96279310/649b0e48-ed2b-4d3f-a937-8fb5a459cd8f)

A better experience would have been if the overload said "I don't know, but here is what I made up".

**Comment by william-liebenberg** [comment]
Cool - great start to using GPT in a custom application.

**Comment by Geordie88** [comment]
While google is good for finding general rules categories, the bot does a great job directing me to specific rules, which as an SSW employee, is what i generally need :D. in short it's awesome

**Comment by Harry-Ross** [comment]
Think that its great - apart from the fact that sometimes GPT has the potential to be a bit too creative with the rules links it returns. 
![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/20507092/3c5ee98b-bf18-4d61-86c3-c1d01d8eb16c)
Was able to answer an obscure rule that isn't regularly used.

**Comment by chrisschultzssw** [comment]
It answered my question pretty well... but it seems to be a fan of Power Platform @calumjs 🤔.

![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/94873295/4e7893ef-6ec6-442f-91e6-13a08a4ef53a)

**Comment by steven0x51** [comment]
I came up with an idea, we could name it "Rulbot"

**Comment by leahy268** [comment]
>Did you find it useful?
Yes, it was great for finding the answers

    >What question did you ask it?
What is the most important thing if you receive a personal breach?

Answer was:
If you receive a personal breach, the most important thing is to immediately contact your SysAdmins for help, as per https://ssw.com.au/rules/security-compromised-password. This is important because a personal breach, such as a hacked Gmail account, can have implications for the company. You should expect the same level of service on a personal breach as you would on a corporate breach. Once you have informed your Systems Administrator of a potential breach, send an "As Per My Conversation" email to document the incident.

This is part of a SugarLearning item which I approve and it got the correct answer much better than most beginning Dev's.

**Comment by JackDevAU** [comment]
I think this is a great idea, but it needs some work! 

It gave me a very generic answer and didn't link to the rule or provide the right template. 
![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/57518417/90b6f30a-cd35-4a27-bfd4-d8fa797d3481)

**Comment by PennyWalker** [comment]
Really cool - just help finding the right Rule will be useful. Google often can't find things on the first go and you have to dig for it. It answered the first two questions pretty well, but the last one was very generic, so not as helpful. It would have been nice to get a yes, or no answer. It did link to the correct Rule but didn't process the data for me.

![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/25000905/536fde55-f81e-437c-ba3d-1613265db1a3)
![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/25000905/4c875207-20e3-46a6-b6f5-104fbc93877c)
![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/25000905/80b15824-3bf7-41f8-af91-f49973a1f029)

**Comment by JeanThirion** [comment]
![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/33501311/1e1ba3f3-4f83-481b-8f05-910823e7a6ee)


I absolutely love this

**Reply by calumjs** [reply]
Perfection

**Comment by piers-sinclair** [comment]
Somewhat useful, it confirmed what I thought.

![Screenshot_20230519_103345_Samsung Internet](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/79821522/7026847c-47e3-4b1b-ab3d-2e3a6914381f)

**Comment by Dhruv-0987** [comment]
This is just amazing. For people who like me who don't like to read anything except documentation and code this is a lifesaver.

**Comment by Ben0189** [comment]
This app is amazing! It helps me find specific SSW rules even when I only remember part of the content. It's a time-saving and convenient tool.

**Comment by bradystroud** [comment]
Seems to work well, lets give it a few more frontends!

1. It could become a little chat thingy that floats on the bottom right of the rules site.

Similar to this

<img width="1680" alt="image" src="https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/38869720/5374bd1b-76c4-4cd4-bb3a-f7bd5f924f7a">

2. Integrate with SophieBot/Make a new Teams bot
3. Mobile app? its currently built in Blazor, so it would be easy drop it into a MAUI app
4. 🤷🏻‍♂️

**Reply by camillars** [reply]
+1 with #2

**Comment by CoffeeBugHub** [comment]
A bit slow but it does give me what I want!
![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/62824904/2aa01f38-dd6a-4ae3-b4b2-00f1f7f3f855)

**Comment by landonmaxwell** [comment]
Very cool

**Comment by UlyssesMaclaren** [comment]
sometimes it's not as opinionated as the actual rules are... e.g. i asked how to do placeholders in an email template, and it 1st gave me the wrong options, then multiple options, and took more drilling down to land on the correct one {{ }}

**Comment by KostaMadorsky** [comment]
That's awesome. 
Really convenient finding rules that way!

**Comment by Abf09** [comment]
Seems very convenient. When I give it the right right context it does answer with an example which saves so much time. For example when I asked it how do I send an absence mail, the answer was very generic however, upon asking how do I send an SSW absence mail it does give the right answer with the correct example.

**Comment by NadeekaJK** [comment]
Like the idea of adding it to rules, I should also try to outsource my remaining Sugar Learning tasks to chat-GPT 
![EvilVillainYawnGIF](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/94897150/d4768996-4408-479f-93d1-863c6e3ac4d5)

**Comment by camillars** [comment]
Very cool and useful - great work @calumjs! ⭐

**Reply by calumjs** [reply]
@MattParkerDev @jackreimers & @Harry-Ross are the heroes, I just watched 😁

**Reply by camillars** [reply]
great work, team!

**Comment by drwharris** [comment]
![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/19632724/e2484871-7604-4789-883b-0cde61723a8b)

**Comment by Hona** [comment]
Its pretty cool, would be cool to shove it into a MAUI Hybrid app & get mobile for free

**Comment by sylhuang** [comment]
![image](https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/23565403/c6cde465-9704-42e4-8989-7a2290884e65)
Though it's a non-existing rule, it provides good content suggestion to include in the rule! 😼
(But it failed to make up more 😂)

**Comment by johnxu16** [comment]
Even if it gives us misleading information sometimes, if we have a general idea of what kind of rules we may have, we would get decent answers.

**Comment by JaneYangssw** [comment]
That's really cool!!!!!!!! Now we SSWers can use our own exclusive GPT!!!

**Comment by jerryluossw** [comment]
That's so cool, it's easier to find the rule I want, but sometimes it will reply to a fake rule, and the rule looks real.

**Comment by Aibono1225** [comment]
Great tool! It would be even better if the accuracy could be improved. 🥰

**Comment by hveraus** [comment]
Seems like a lot of broken links. I think it use another source if RuleGPT can not find answer in the rules database. 
Should we limit it only search for rules?

Love the dark mode!

**Comment by KristenHu** [comment]
It's cool to have a rulesGPT, I'm also looking forward to other functions beyond searching rules.😍

**Comment by AnkyChe** [comment]
Really cool! And I can use it to generate email template which I think is really useful.

**Comment by tino-liu** [comment]
Still gotta figure out the right prompts to ask RulesGPT, but it does seem like a really great integration! Now I don't have to manually search for SSW Rules.

**Comment by AttackOnMorty** [comment]
The rules bot is awesome!  It would be great if we can directly access the rules bot on our Rules page. Similar to the chatbot on our official website.

<img width="1436" alt="image" src="https://github.com/SSWConsulting/SSW.Rules.GPT.Blazor/assets/37203901/afc5e197-0cd9-40c7-b7c7-36d240875ea4">

**Comment by SofieHong** [comment]
Amazing creativity! 🥳 Although it got stuck when I asked the second question 🫢
There's no denying that this is the strongest rules generator! It will greatly reduce the time we spend looking for a rule, allowing us to focus more energy on the content.

**Comment by AlvinShenSSW** [comment]
It's handy to find rules, but sometimes it is not working in China. We need to Chinafy it.

**Comment by ozair-ashfaque** [comment]
Awesome, great effort Callum

**Comment by BrookJeynes** [comment]
another useful tool for finding rule information

**Comment by PothieuG** [comment]
The chat helped me find two rules that I couldn't locate on my own!
I think I'll be using this tool very often. Very nice tool :)

**Comment by TobyChurches** [comment]
A really nice feature! Definitely makes it easier to locate the information in our extensive set of rules!

**Comment by Freego1783** [comment]
Sometimes you're sure that a certain rule exists but you can't find it, now RulesGPTBot-Turbo-Pro-XL will make the search easier !

**Comment by sethdaily** [comment]
It seems to be helpful when looking for a specific rule. It got confused when I was asking it to list multiple rules that include the same keywords.

**Comment by jackreimers** [comment]
The rules bot is very useful for finding information on rules quickly when you may not know the specific name or topic of the rule to find it by
